### PR TITLE
add github management policies dir

### LINKED ISSUE #153: TAC question: where should ossf github org policies/tools live?
Hi folks!

I asked this on Slack but figured here would be better for the long term.

I'm going to be writing up some repo management policies (via PR, seeking TAC approval, ofc) and am trying to figure out the best place for them to live. A top-level dir in this repo seems appropriate?

Also, I've written a very rough tool in the (currently private, but will be made public once a home is decided) https://github.com/ossf/github-org-access-scraper repo to be able to audit how many repos still have individuals (as opposed to teams) with access - should that also live in this repo, instead of its own?

### PR DESCRIPTION
**Pull request by ljharb**
Closes #153, assuming the tooling should live in a separate repo.

This is just an initial doc to get things started; happy to take any and all feedback.

### CODE REVIEW

File: README.md
Comment (ctcpip) [inline README.md:25]: merge conflict?

Comment (ljharb) [reply README.md:25]: oh whoops, thanks, this file isn't supposed to be included :-)

File: policies/access.md
Comment (tpepper) [inline policies/access.md:32]: This feels different than many open source projects. Especially for a relatively new and also broad umbrella foundation, I worry that this freely given perpetual membership even if only a weak badging, might encourage confusion on what is or is not officially part, sponsored by, under consideration within the auspices of the foundation. That type of confusion and contention has already been a bit of a problem for the OpenSSF org. 

I personally would prefer a simple and collaborative open membership. But I also want clarity of mission and propose and that to be easily discovered and difficult to confuse. 

Perhaps this is less of an issue if leadership (ie: certain team memberships in the org?) roles are clearly marked. But GitHub team memberships tend to be opaque. I’m not sure if there is maybe an org setting which would allow team members be viewed by non org members.  Or if not…alternatively if anybody can get org membership, I guess anybody could ask for org membership to be able to see team membership?  Or GH team members defined by in repo yaml configs?  Still not great UX though.

Comment (ljharb) [reply policies/access.md:32]: @tpepper OpenSSF is not itself an open source project, and this is how TC39 operates, which I think is more analogous.

I agree that it's not great UX that the only way to be on a team is to be in the org, but it seems a worthy tradeoff for me.

Comment (steiza) [inline policies/access.md:32]: Currently, it looks like members of https://github.com/ossf can create public and private repositories. We should probably audit https://github.com/organizations/ossf/settings/member_privileges before opening up membership as described in this document.

**Review by steiza — COMMENTED** [review]
I love the proposal to use teams to manage access!

I have some questions about how to open up membership to the OpenSSF organization broadly, and how to manage the consequences of that.

File: policies/access.md
Comment (ljharb) [reply policies/access.md:32]: It seems reasonable to me to restrict repo creation to OpenSSF staff and the TAC, if there’s agreement on that.

File: policies/access.md
Comment (ljharb) [reply policies/access.md:30]: This document doesn’t propose automating this at all, so the answer is “manually” - you need to ask someone who can direct your request to OpenSSF staff, and they can resend your invite as well.

Comment (steiza) [reply policies/access.md:30]: Fair enough! Is there an OpenSSF staff email address we should list here? Or is that out of scope of the policy document?

Comment (ljharb) [reply policies/access.md:30]: I think that's out of scope of the document; it's pretty easy to reach folks in the Slack :-)

**Review by steiza — COMMENTED** [review]
Minor: there's a bunch of places I'd like us to switch `Github` to the official `GitHub`.

Other than that, I'm a strong supporter of organizing permissions via teams, and I'm willing to give membership to github.com/ossf for anyone who asks a try.

File: policies/access.md
Comment (steiza) [inline policies/access.md:30]: From https://github.com/ossf/tac/issues/164#issuecomment-1549952618, can we break this section out of this pull request, and give it its own pull request?

We'd like to start the team / role-based access control as soon as possible to help with things like https://github.com/ossf/tac/issues/161 and https://github.com/ossf/tac/issues/162. 

The TAC discussion yesterday on `GitHub Org Membership` was generally in favor, but I think folks would like to ask more clarifying questions and have a longer discussion around this point. But in the meantime, we don't wan to block the team / role-based access control work. Thanks!

Comment (ljharb) [reply policies/access.md:30]: hmm, that's not how i read @AevaOnline's comment - i read it as needing a separate task to see what could be *built on top of* this section.

The team-based work is blocked on being able to freely add people to the org - and I've been doing that for the last few months as part of my audit already.

Comment (steiza) [reply policies/access.md:30]: The way I interpret "membership in the GitHub org should be freely given" is that we would be required to add anyone who asks to https://github.com/ossf, even if they don't end up on any GitHub org teams.

Without this section, my understanding is we would still be able to add people to https://github.com/ossf in order to complete the team-based work.

Comment (ljharb) [reply policies/access.md:30]: “should” doesn’t obligate us, but yes it’s also fine to have someone in the org on no teams - that makes them easily addable *to* teams.

Fine to remove this section though if the TAC feels it should be deferred.

Comment (ljharb) [reply policies/access.md]: This is done; I'll open the second PR once this one lands.

**Review by steiza — APPROVED** [review]
Looks good to me; thanks!

**Review by bobcallaway — APPROVED** [review]

**Review by lehors — APPROVED** [review]
LGTM

**Review by hythloda — APPROVED** [review]

File: policies/access.md
Comment (ctcpip) [inline policies/access.md:34]:
```suggestion
- There are few settings that justify Admin access over Maintain, so prefer Maintain.
```

File: policies/access.md
Comment (ctcpip) [inline policies/access.md:35]:
```suggestion
- Explicit Read access has an advantage: users with Read can be assigned to issues and requested as PR reviewers even if they're not the author.
```

File: policies/access.md
Comment (ctcpip) [inline policies/access.md:9]:
```suggestion
Teams will be nested, using parent/child relationships, as needed. A child team will implicitly have any privileges its parent team does, so the most deeply nested team should be the one with the most privileges.
```

**Review by ctcpip — CHANGES_REQUESTED** [review]
minor grammar/punctuation issues

**Review by di — APPROVED** [review]

**Review by AevaOnline — APPROVED** [review]
LGTM!

File: policies/access.md
Comment (hythloda) [reply policies/access.md]: This is NOT done correctly. I am not sure how to allow certain teams to create repos.
In https://github.com/organizations/ossf/settings/member_privileges it is not team-based. I would love a learning session on this from someone who knows how to do this.

File: policies/access.md
Comment (jeffmendoza) [inline policies/access.md:28]: Following PLP(below) projects will need teams for each access level (eg scorecard-read, scorecard-traige, scorecard-write, scorecard-maintain)

File: policies/access.md
Comment (jeffmendoza) [inline policies/access.md:5]: How to we enforce this?

Allstar has a policy to nag if any individuals have write/admin access, we could use this.

File: policies/access.md
Comment (jeffmendoza) [inline policies/access.md:34]: Looks like Maintain doesn't allow setting up branch protection. A lot of or repos don't have this setup Will org admins set this up, if we don't have Admin access to our own repos?

File: policies/access.md
Comment (ljharb) [reply policies/access.md:28]: In the extreme yes, but we'd only create these as needed, since most repos don't need this granularity atm.

File: policies/access.md
Comment (ljharb) [reply policies/access.md:5]: We already have an auditing tool; see https://github.com/ossf/github-org-access-scraper.

File: policies/access.md
Comment (ljharb) [reply policies/access.md:34]: Yes; org admins (the PM team, primarily) will be available to do any repo administration that repo owners lack access to do.

File: policies/access.md
Comment (ljharb) [reply policies/access.md]: @hythloda the mechanism we're currently using is that only org owners can create repos, and all TAC members and some PMs are org owners. It indeed can't be team-based.

**Review by ctcpip — APPROVED** [review]

**Comment by ljharb** [comment]
I believe all comments have been addressed and this seems to have sufficient approvals.

I'll land this later today if there's no further feedback/objections.

File: policies/access.md
Comment (jeffmendoza) [reply policies/access.md:28]: I don't think this is just an extreme case. Take a look at the BP-WG group:
https://github.com/orgs/ossf/teams/wg-best-practices/members
https://github.com/orgs/ossf/teams/wg-best-practices/repositories

All those people who had some kind of access to any of the 6 repos now have write/maintain access to all of them. In an effort to reduce the manual configuration of groups. repos and access-levels will be consolidated and people will be getting access they don't need and/or shouldn't have.

Comment (ljharb) [reply policies/access.md:28]: Sure, that's a fair point. However, when I created that specific team i was told that all members should have Write. There's a tradeoff here balancing convenience and "least privilege", and since Write is a pretty non-dangerous privilege, it seemed like a worthy one.

If we end up needing to do this for the majority of repos, then we should absolutely document it as the default approach, but we're not there yet today.

File: policies/access.md
Comment (jeffmendoza) [reply policies/access.md]: This doesn't seem like a tenable solution. If encouraging project owners to not have admin on the repos, they really need an easy way to read settings, and an automated way to apply changes. 
https://gitlab.eclipse.org/eclipsefdn/security/otterdog is a tool we could use to do this, for example.

Comment (ljharb) [reply policies/access.md]: This is definitely something we can iterate on; in my experience managing large github orgs, maintainers rarely need Admin more than a handful of times over the lifetime of a repo, and as long as the friction is minimal when it's needed, it's "fine".

Note also that the language here is "prefer"; if anyone feels strongly that they want admin over a repo, then imo they can have it.

**Review by SecurityCRob — APPROVED** [review]
+1

File: policies/access.md
Comment (jeffmendoza) [reply policies/access.md:28]: > There's a tradeoff here balancing convenience and "least privilege"

This is exactly my point. Without granular teams per repo and access level, we are reducing our security and going against PLP. Without some sort of automation or config-as-code way to manage the teams, convenience will undermine the goals of the teams mandate.

Comment (ljharb) [reply policies/access.md:28]: I think that is something we can iterate on as well - automation is worth considering but in practice is often overkill, and the goal here isn't strict PLP, but a safer balance.

File: policies/access.md
Comment (jsoref) [inline policies/access.md:27]:
```suggestion
    This is the parent team for projects. Every project (e.g. scorecard, AO) should have a subteam contained within this one.
```

File: policies/access.md
Comment (joshuagl) [inline policies/access.md:3]: One commonly encountered downfall of using GitHub teams is that the teams and their members are not publicly visible. Teams have two states, [visible and secret](https://docs.github.com/en/organizations/organizing-members-into-teams/about-teams#team-visibility). Visible teams can be viewed and mentioned **by organisation members**. Secret teams can be viewed and mentioned only by people on the team and organisation owners. 

This observation need not be a blocker for this effort, but we should likely document it here and be more aware of it in how we communicate governance. For example, [linking to teams in README's](https://github.com/ossf/wg-securing-software-repos/issues/15) isn't very useful for the majority of folks reading the README.

Comment (joshuagl) [reply policies/access.md:3]: I may be wrong here, I can see 9 of the 44 teams in this view https://github.com/orgs/ossf/teams which suggests there's a third visibility state for teams.

Comment (jsoref) [reply policies/access.md:3]: I'm pretty sure I don't see any teams, just repositories. Try browsing that page using incognito?

Comment (ljharb) [reply policies/access.md:3]: I’m aware of this, yes - the long term intention is to automate things so that the readme will list the members of a team, and stay in sync with it. If folks think it’s worth documenting how github works here that’s fine, but I’m not going to hold up this PR on it.

Comment (jsoref) [reply policies/access.md:3]: Since one of your reviewers tripped on this behavior, and I doubt most GitHub users are well versed in the details, and it sounds like you have quite a few members, it's probably a good idea to add some text to mention all of these points.

Comment (ctcpip) [reply policies/access.md:3]: oppose re-documenting GH here, or anywhere.  at most, linking to [the relevant documentation](https://docs.github.com/en/organizations/organizing-members-into-teams/about-teams#team-visibility) is sufficient

File: policies/access.md
Comment (ljharb) [reply policies/access.md]: Filed #171

**Pull request merged**
